### DISCUSSION TITLE
Organizing scripts

### DISCUSSION THREAD

**Discussion by JordanMarr** [grate-devs/grate#269] [General]
In the past, I have organized scripts that I ran manually in their own feature related folders.
Would it make sense to organize grate scripts in separate feature folders, each having its own `up` folder (and others as needed)?
Or should I strive to continually add all migration scripts together and just continually add new scripts as needed?

**Comment by erikbra** [comment]
I would recommend ordering the scripts. I think I would organize them the other way around, putting feature folders inside the `up` folder, and name them alphabetically (e.g. prefixed with an increasing number, a tracking/ticket number, etc), to keep the ordering.

The stored procedures, etc, can also be organized similarly, if it makes sense, but they can of course change over time, and maybe the change is related to a different feature, so it might be difficult as time goes by.

There are a lot of ways to organize the scripts, of course, but, personally, I think, if you'd like some folder structure, I'd organize them as follows:

up/0001-featureA
up/0002-featureB
up/0003-featureA.bugfix1
up/0004-featureC

etc

**Comment by JordanMarr** [comment]
Ah ok, I didn't realize you could nest folders inside the special folders. 
Thanks, that seems like a great way to do it.

**Reply by erikbra** [reply]
Good :) Hope it works out for you!
